caffe2/operators/experimental/c10/cpu/add_cpu.cc

```diff
@@ -11,24 +11,27 @@ namespace {
 
 template <class DataType>
 void add_op_cpu_impl(
-    const Tensor& A,
-    const Tensor& B,
-    Tensor* C,
+    const C10Tensor& A_,
+    const C10Tensor& B_,
+    const C10Tensor& C_,
     bool legacy_broadcast,
     int axis,
     BaseContext* context) {
+  Tensor A(A_);
+  Tensor B(B_);
+  Tensor C(C_);
   const DataType* A_data = A.template data<DataType>();
   const DataType* B_data = B.template data<DataType>();
   std::vector<int> A_dims;
   std::vector<int> B_dims;
 
   if (legacy_broadcast) {
     CAFFE_ENFORCE_NE(
-        C,
-        &B,
+        C.getIntrusivePtr(),
+        B.getIntrusivePtr(),
```

Comment:
`at::Tensor` has `is_same` for testing this. We should do that, instead of checking for equality on the intrusive pointers. You shouldn't call `getIntrusivePtr` unless you have no choice, and checking if the tensors are the same is not a good reason!

Reply:
I'll add `caffe2::Tensor::is_same` in a separate diff.

Reply:
https://github.com/pytorch/pytorch/pull/15407